runtime/make/Makefile.runtime.head

```diff
@@ -39,6 +39,7 @@ CHPL_MAKE_HOST_TARGET = --target
 endif
 include $(RUNTIME_ROOT)/../make/Makefile.base
 
+RANLIB ?= ranlib
```

Comment:
Not to belabor this issue too much since `make` is like black magic. Without `RANLIB` disappearing, I would have never found out that I set one of the `CHPL_*_COMPILER` variables to an invalid value. (Of course, once I fixed that, `RANLIB` was fine.) What is confusing to me is why didn't the other builtin variables like `AR` also get affected. I'm mildly concerned that by manually setting `RANLIB`, this may someday hide a different problem.

Reply:
`make` sets default values for a number of standard Makefile variables like CC, CXX, AR, etc.  I couldn't find any indication that there is an equivalent for RANLIB which is why I set a default here.  Hopefully the warning generated by this PR will protect against such mis-set variables in the future rather than relying on a secondary effect like RANLIB not being set (i.e., I'd rather have users use the warning as a cue that something's wrong rather than having to debug the fact that RANLIB was turning into an empty string, which is pretty subtle / advanced).

Reply:
PS - we could also consider turning the warning into an error to make it a bit more "in your face" and hard to ignore and I'm not deeply opposed to that.  The one reason to keep it a warning is that it gives a (roundabout) way for someone to build using their default CC, CXX, etc. variables for better or worse (e.g., I'm thinking about issue #12777 and how this arguably provides a lame way to have such variables be respected even though I'm suspicious that this is deeply useful).